agent/command/s3_put.go

```diff
@@ -571,13 +571,15 @@ func (s3pc *s3put) remoteFileExists(remoteName string) (bool, error) {
 		AwsSessionToken: s3pc.AwsSessionToken,
 		Region:          s3pc.Region,
 	}
-	_, err := pail.GetHeadObject(requestParams)
+	_, err := pail.GetHeadObject(ctx, requestParams)
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == notFoundError {
-			return false, nil
-		} else {
-			return false, errors.Wrapf(err, "getting head object for remote file '%s'", remoteName)
+		var smithyErr smithy.APIError
+		if errors.As(err, &smithyErr) {
+			if smithyErr.ErrorCode() == notFoundError {
```

Comment:
Would we consider using [s3.NotFound.ErrorCode()](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3/types#NotFound.ErrorCode) here now instead of defining our own constant?

Reply:
It would be a little awkward because the NotFound error object has to be stored in a variable before it can be used:
```go
notFoundErr := s3Types.NotFound{}
if smithyErr.ErrorCode() == notFoundErr.ErrorCode() {
}
```

Anyways, this error check should go away after DEVPROD-9646 is completed.